zeppelin-web/src/components/notebookListDataFactory/notebookList.datafactory.js

```diff
@@ -13,14 +13,28 @@
  */
 'use strict';
 
-angular.module('zeppelinWebApp').factory('notebookListDataFactory', function() {
+angular.module('zeppelinWebApp').factory('notebookListDataFactory', function($location) {
   var notes = {};
 
   notes.list = [];
 
   notes.setNotes = function(notesList) {
+  	if(notes.list.length && notes.list.length+1===notesList.length){
+		var newNoteID = getNewNoteID(notesList);
```

Comment:
is there a chance for a race condition if multiple notebooks are being added (say from multiple people)?